src/sentry/search/snuba/executors.py

```diff
@@ -472,6 +478,323 @@ def query(
+                return (
+                    0
+                )  # Maybe check for 0 hits and return EMPTY_RESULT in ::query? self.empty_result
+            else:
+                filtered_count = group_queryset.filter(
+                    id__in=[gid for gid, _ in snuba_groups]
+                ).count()
+
+                hit_ratio = filtered_count / float(snuba_count)
+                hits = int(hit_ratio * snuba_total)
+        elif group_ids:
+            return len(snuba_groups)
+
+        return hits
+
 
 class SnubaOnlyQueryExecutor(AbstractQueryExecutor):
-    pass
+    TABLE_ALIAS = "events."
+    ISSUE_FIELD_NAME = TABLE_ALIAS + "issue"
+    logger = logging.getLogger("sentry.search.snubagroups")
+
+    # TODO: Define these variables using table alias somehow?
+    # Since I think that is the only difference, other than issue_only_fields having less items
+    dependency_aggregations = {"priority": ["events.last_seen", "times_seen"]}
+    issue_only_fields = set(
+        ["query", "bookmarked_by", "assigned_to", "unassigned", "subscribed_by"]
+    )
+    sort_strategies = {
+        # TODO: If not using environment filters, could these sort methods use last_seen and first_seen from groups instead? so only add prefix conditionally?
+        "date": "events.last_seen",
+        "freq": "times_seen",
+        "new": "events.first_seen",
+        "priority": "priority",
+    }
+
+    aggregation_defs = {
+        "times_seen": ["count()", ""],
+        "events.first_seen": ["multiply(toUInt64(min(events.timestamp)), 1000)", ""],
+        "events.last_seen": ["multiply(toUInt64(max(events.timestamp)), 1000)", ""],
+        "priority": ["toUInt64(plus(multiply(log(times_seen), 600), `events.last_seen`))", ""],
+        "total": ["uniq", ISSUE_FIELD_NAME],
+    }
+
+    @property
+    @abstractmethod
+    def dataset(self):
+        return snuba.Dataset.Groups
+
+    def _transform_converted_filter(
+        self, search_filter, converted_filter, project_ids=None, environment_ids=None
+    ):
+        converted_name = search_filter.key.name
+
+        special_date_names = ["active_at", "first_seen", "last_seen"]
+        if search_filter.key.name in special_date_names:
+            # Need to get '2018-02-06T03:35:54' format out of 1517888878000 format
+            datetime_value = datetime.datetime.fromtimestamp(converted_filter[2] / 1000)
+            datetime_value = datetime_value.replace(microsecond=0).isoformat().replace("+00:00", "")
+            converted_filter[2] = datetime_value
+
+        # TODO: There is a better way to do this...the issue is that the table/alias is forked on environments, which can't be used in constrain_column_to_dataset
+        if search_filter.key.name in ["first_seen", "last_seen"]:  # , "first_release"]:
+            if environment_ids is not None:
+                table_alias = "events."
+                # return None, None #??? remove as a filter, since we are going to pre/post filter?
```

Comment:
Not following why would you ever use events.first_seen if there is an environment provided

Reply:
I now provide a way to remove this as a condition.

We no longer use events.first_seen (except for sorting) - we just don't apply this filter to the snuba query, and will apply it to a postgres query, which will be applied via pre/post filtering.